File: flutter/function.dart
// 无关键字
// 有可选命名参数
// 有默认参数值

//

// 无参数，无返回值
void Void(){
  print('void');
}
// 有可选命名参数和默认值，有返回值
int Test(int a, {bool b=false, String? s}){
  print([a, b, s]);
  return a;
}

void main() {
  Void();          // 结果： void
  var t = Test(1); // 结果： [1, false, null]
  print(t);        // 结果： 1
}

File: php/php-log/function.php
<?php
// 由关键字```function```声明
// 有默认参数值
// 有可变长参数(不定长参数)

//

// 无参数，无返回值
function Void(): void {
    echo "void\n";
}

// 可变长参数(不定长参数)
function Names(string ...$s){
    var_dump($s);
}

// 有参数有默认参数值，有返回值
function Test(int $a, bool $b=false, string $s='str...'): int{
    echo $a, ',',$b, ',', $s, "\n";
    return $a;
}

Void();               // 结果： void
Names('a', 'b', 'c'); // 结果：array(3) {[0]=>string(1) "a" [1]=>string(1) "b" [2]=>string(1) "c"}
$t = Test(1);        // 结果： 1,,str...
echo "$t\n";         // 结果： 1
